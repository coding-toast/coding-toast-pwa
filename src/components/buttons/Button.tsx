import * as React from 'react'
import { Link } from 'gatsby'

interface ButtonProps {
  toUrl: string
  buttonText: string
  className: string
}

const Button: React.FC<ButtonProps> = props => {
  const { toUrl, buttonText, className } = props

  return (
    <Link to={toUrl} className={className}>
      {buttonText}
    </Link>
  )
}

export default Button
