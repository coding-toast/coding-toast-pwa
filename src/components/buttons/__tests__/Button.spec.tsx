import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from '../Button'

describe('<Button />', () => {
  it('renders correctly', () => {
    const component = shallow(<Button toUrl="/home" buttonText="test" className="test" />)
    expect(toJson(component)).toMatchSnapshot()
  })
})
