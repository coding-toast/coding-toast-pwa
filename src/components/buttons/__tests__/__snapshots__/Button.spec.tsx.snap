// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<mockConstructor
  className="test"
  to="/home"
>
  test
</mockConstructor>
`;
